backend/courses/views.py

```diff
@@ -520,7 +520,8 @@ def get_all_friendships(self):
 
     def post(self, request):
         sender = request.user
-        recipient = get_object_or_404(User, username=request.data.get("pennkey"))
+
+        recipient = get_object_or_404(User, username=request.data.get("pennkey").lower())
```

Comment:
You should probably build in some logic here to handle the case where `request.data.get("pennkey")` returns `None`; this probably shouldn't happen based on input validation but we wouldn't want this to crash because of NoneType exception. (same with the other case that you modified)

You might also want to add some tests for unexpected inputs if they're not already there